src/legacy/core_plugins/vis_type_timeseries/server/routes/post_vis_schema.js

```diff
@@ -0,0 +1,218 @@
+  steps: numberIntegerOptional,
+  terms_field: stringOptionalNullable,
+  terms_order_by: stringOptionalNullable,
+  terms_size: stringOptionalNullable,
+  terms_direction: stringOptionalNullable,
+  terms_include: stringOptionalNullable,
+  terms_exclude: stringOptionalNullable,
+  time_range_mode: stringOptionalNullable,
+  trend_arrows: numberOptional,
+  type: stringOptionalNullable,
+  value_template: stringOptionalNullable,
+  var_name: stringOptionalNullable
+});
+
+export const visPayload = {
+  filters: arrayNullable,
+  panels: Joi.array()
+    .items(Joi.object({
+      annotations: Joi.array()
+        .items(annotationsItems).optional(),
+      axis_formatter: stringRequired,
+      axis_position: stringRequired,
+      axis_scale: stringRequired,
+      axis_min: stringOptionalNullable,
+      axis_max: stringOptionalNullable,
+      bar_color_rules: arrayNullable.optional(),
+      background_color: stringOptionalNullable,
+      background_color_rules: Joi.array()
+        .items(backgroundColorRulesItems).optional(),
+      default_index_pattern: stringOptionalNullable,
+      default_timefield: stringOptionalNullable,
+      drilldown_url: stringOptionalNullable,
+      drop_last_bucket: numberIntegerOptional,
+      filter: Joi.alternatives(
+        stringOptionalNullable,
+        Joi.object({
+          language: stringOptionalNullable,
+          query: stringOptionalNullable
+        })),
+      gauge_color_rules: Joi.array()
+        .items(gaugeColorRulesItems).optional(),
+      gauge_width: [stringOptionalNullable, numberOptional],
+      gauge_inner_color: stringOptionalNullable,
+      gauge_inner_width: Joi.alternatives(stringOptionalNullable, numberIntegerOptional),
+      gauge_style: stringOptionalNullable,
+      gauge_max: stringOptionalNullable,
+      id: stringRequired,
+      ignore_global_filters: numberOptional,
+      ignore_global_filter: numberOptional,
+      index_pattern: stringRequired,
+      interval: stringRequired,
+      isModelInvalid: Joi.boolean().optional(),
+      legend_position: stringOptionalNullable,
+      markdown: stringOptionalNullable,
+      markdown_scrollbars: numberIntegerOptional,
+      markdown_openLinksInNewTab: numberIntegerOptional,
+      markdown_vertical_align: stringOptionalNullable,
+      markdown_less: stringOptionalNullable,
+      markdown_css: stringOptionalNullable,
+      pivot_id: stringOptionalNullable,
+      pivot_label: stringOptionalNullable,
+      pivot_type: stringOptionalNullable,
+      pivot_rows: stringOptionalNullable,
+      series: Joi.array()
+        .items(seriesItems).required(),
+      show_grid: numberIntegerRequired,
+      show_legend: numberIntegerRequired,
+      time_field: stringOptionalNullable,
+      time_range_mode: stringOptionalNullable,
+      type: stringRequired,
+    })),
+  // general
+  query: Joi.array()
+    .items(queryObject)
+    .allow(null)
+    .required(),
+  state: Joi.object({}).required(),
```

Comment:
I did a _quick_ search of the code, and it looks like the only expected property here is `sort`:
https://github.com/elastic/kibana/blob/5887526723dea443cab43a5cb95f863cf13aa344/src/legacy/core_plugins/vis_type_timeseries/server/lib/vis_data/request_processors/table/pivot.js#L28

Are you aware of others? If not, we should be explicit about the set of allowed keys.

Reply:
Thanks @legrego , it was a huge payload to unpack and some of these items may have fallen through the cracks.

Reply:
Added it to the validation.

Reply:
@TinaHeiligers it was a massive payload -- thanks so much for taking care of this!

@flash1293 thanks!